localize dates according to document lang in NL, DE and EN (default)

gronke opened the pull request:
Localizes dates in Dutch, German and English (default) with custom functions.

Hardcoding the month names helps to avoid fiddling with Saxon and Java.

gronke committed:
localize dates according to document lang in NL, DE and EN (default)

gronke committed:
comment reason in xslt/dates.xslt

gronke committed:
support English dates in UK and US (default) format

syrosh reviewed (approved):
LGTM

I think it's fine to go ahead with hard coded values. Maybe adding the Java way if we need to support more localization.

gronke committed:
use localized dateTime format in version history

PeterMosmans reviewed (changes requested):
Currently, as it stands, this errors out if there's no date (yet) set:
```
Error at char 36 in xsl:value-of/@select on line 321 column 106 of meta.xslt:
  XPST0008  Variable date has not been declared (or its declaration is not in scope)
  ```
  I'm thinking of whether we can support a default (?) language as fallback, in order to ensure backwards compatibility?

gronke committed:
fix undeclared date variable and add language fallback

Use @date attribute instead of undeclared $date variable in version
history, and use the -tbd variant to handle missing dates gracefully.
Default lang parameter to 'en' when xml:lang is not set.
